web/modules/custom/dpl_event/src/EventSubscriber/EventSeriesRedirect.php

```diff
@@ -42,15 +50,23 @@ public function checkEventSeriesRedirect(RequestEvent $event): void {
 
     $request = $event->getRequest();
     $route_name = $request->attributes->get('_route');
-    $eventSeries = $request->attributes->get('eventseries');
+    $event_series = $request->attributes->get('eventseries');
 
-    if ($route_name !== 'entity.eventseries.canonical' || !$eventSeries instanceof EventSeries || $eventSeries->getInstanceCount() > 1) {
+    if ($route_name !== 'entity.eventseries.canonical' || !$event_series instanceof EventSeries) {
       return;
     }
 
-    $eventInstances = $eventSeries->get('event_instances')->referencedEntities();
-    if (count($eventInstances) === 1 && $eventInstances[0] instanceof EventInstance) {
-      $response = new RedirectResponse($eventInstances[0]->toUrl()->toString());
+    $upcoming_ids = $this->reoccurringDateFormatter->getUpcomingEventIds($event_series);
+
+    if (count($upcoming_ids) > 1) {
+      return;
+    }
+
+    $event_instance_id = reset($upcoming_ids);
+    $event_instance = $this->entityTypeManager->getStorage('eventinstance')->load($event_instance_id);
```

Comment:
I am unsure about this change.

1. Code-wise: What happens if there are no upcoming instances? `$upcoming_ids` will be empty, `reset()` on an empty array with return `false` but what happens when `false` is passed to `load()`?
2. Functionally I still think it is appropriate to redirect from series with a single instance to the actual instance even if it is in the past.

Reply:
1) Good point - I've updated line 62 to: `if (count($upcoming_ids) == 1) {`
2) I'm not sure I agree with you here - the whole point of the task is that expired events should be treated as second-class content - they should not show up in various lists, and only accessed if the user has a direct link to them. 
If an eventseries only has expired events, the instances wont show up in the list, and then i wouldnt expect a redirect to happen